### ISSUE TITLE
ReduceLROnPlateau needs to set "frequency" when monitoring validation metrics

### ISSUE THREAD

**Issue by cowwoc** [PyTorchLightning/pytorch-lightning#9815]
I want to change the documentation found at https://pytorch-lightning.readthedocs.io/en/latest/api/pytorch_lightning.core.lightning.html#pytorch_lightning.core.lightning.LightningModule.configure_optimizers to highlight the fact that `ReduceLROnPlateau` needs to set `frequency` if it is monitoring validation metrics and `Trainer.check_val_every_n_epoch` is not equal to 1.

I'd change:

```
# The ReduceLROnPlateau scheduler requires a monitor
def configure_optimizers(self):
    optimizer = Adam(...)
    return {
        "optimizer": optimizer,
        "lr_scheduler": {
            "scheduler": ReduceLROnPlateau(optimizer, ...),
            "monitor": "metric_to_track",
        },
    }
```

to

```
# The ReduceLROnPlateau scheduler requires a monitor
def configure_optimizers(self):
    optimizer = Adam(...)
    return {
        "optimizer": optimizer,
        "lr_scheduler": {
            "scheduler": ReduceLROnPlateau(optimizer, ...),
            "monitor": "metric_to_track",
            "frequency": "value of trainer.check_val_every_n_epoch"
        },
    }
```

If `ReduceLROnPlateau` is monitoring the validation loss and `frequency` is not set, then `ReduceLROnPlateau` will not be able to find the metric.

Can you please let me know which file I need to fork to make this change?

**Comment by rohitgr7** [comment]
you need to make the changes here: https://github.com/PyTorchLightning/pytorch-lightning/blob/9e621b451f521397b2703069d2faa8956507a06f/pytorch_lightning/core/lightning.py#L1207
